tests/test_pipelines_common.py

```diff
@@ -394,9 +394,10 @@ def test_save_load_optional_components(self):
         pipe.to(torch_device)
         pipe.set_progress_bar_config(disable=None)
 
-        # set all optional components to None
+        # set all optional components to None and update pipeline config accordingly
         for optional_component in pipe._optional_components:
             setattr(pipe, optional_component, None)
+        pipe.register_modules(**{optional_component: None for optional_component in pipe._optional_components})
```

Comment:
What's this for? 

@patrickvonplaten could you also take a look?

Reply:
`test_save_load_optional_components` fails when optional components in a pipeline are actually initialized before being set to `None` here, because the change is not registered in the pipeline's `_internal_dict`.

Reply:
What situation leads to this case? I am asking this because we didn't have to modify utils in the initial Pix2Pix Zero tests.

Reply:
Initializing the dummy pipeline's `inverse_scheduler` component to a new `DDIMInverseScheduler()` instead of leaving it as `None` (before running this test) is the root cause of the failure here and in #2837.

Reply:
Hmm. I am not entirely sure if this is the right / best way to handle it. I'll wait for @patrickvonplaten's comment.

Reply:
@clarencechen could we maybe just overwrite `test_save_load_optional_components` directly in the pipeline testing file: https://github.com/huggingface/diffusers/pull/2942/files#diff-4ddfd60f39964fe9819918608485908a540de4362dfe26f138a6773f3e7bc3cc instead? I'm not very keen on changing the general testing logic here if that's ok